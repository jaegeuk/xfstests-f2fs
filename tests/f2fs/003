#! /bin/bash
# FS QA Test No. f2fs/003
#
# i_size recovery test
#
#-----------------------------------------------------------------------
# Copyright (c) 2014 Jaegeuk Kim.  All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
    cd /
    rm -f $tmp.*
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter

_supported_fs generic
_supported_os Linux
_require_scratch

testfile=$SCRATCH_MNT/testfile

_scratch_mkfs > /dev/null 2>&1
_scratch_mount

$XFS_IO_PROG -t -f \
	-c "pwrite -S 0x58 0 4096"					\
	-c "fsync"							\
	-c "falloc -k 4k 4k"						\
	-c "fsync"							\
$testfile | _filter_xfs_io

md5sum $testfile | _filter_scratch
src/godown $SCRATCH_MNT | tee -a $seqres.full
_scratch_cycle_mount
md5sum $testfile | _filter_scratch

rm $testfile

status=0
exit
