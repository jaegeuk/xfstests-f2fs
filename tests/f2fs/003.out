QA output created by 003
wrote 4096/4096 bytes at offset 0
XXX Bytes, X ops; XX:XX:XX.X (XXX YYY/sec and XXX ops/sec)
62c6c6286e69526fd15cb97eb1644651  SCRATCH_MNT/testfile
62c6c6286e69526fd15cb97eb1644651  SCRATCH_MNT/testfile
